clang_bind/cmake_frontend.py

```diff
@@ -220,29 +223,35 @@ def _set_targets_from_codemodel(self):
                 target_obj = Target(Path(self.reply_dir, target_file))
                 self.targets[target_obj.get_name()] = target_obj
 
-    def get_dependencies(self, target=None):
-        """Get dependencies of the target(s).
+    def get_library_targets(self):
+        """Get all library targets' names.
 
-        :param target: Target to get the dependencies, defaults to None
-        :type target: str, optional
-        :return: Dependencies of the target(s).
-        :rtype: dict
+        :return: Library targets.
+        :rtype: list
         """
-        targets = [self.targets.get(target)] if target else self.targets.values()
-        return {
-            target.get_name(): list(
-                map(lambda x: x.split("::")[0], target.get_dependencies())
-            )
-            for target in targets
-        }
+        library_targets_objs = filter(
+            lambda target: target.get_type() == "SHARED_LIBRARY", self.targets.values()
```

Comment:
What about STATIC build?

Reply:
Didn't see it in `pcl`, so didn't get to test how it is.

Reply:
`PCL_SHARED_LIBS` can be used to choose between shared and static library for pcl

Reply:
Can you please provide a reference link or example?

Reply:
`cmake -DPCL_SHARED_LIBS:BOOL=TRUE` or similar